src/components/AppSettingsDialog/AppSettingsDialog.vue

```diff
@@ -0,0 +1,308 @@
+ -
+ - This program is distributed in the hope that it will be useful,
+ - but WITHOUT ANY WARRANTY; without even the implied warranty of
+ - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ - GNU Affero General Public License for more details.
+ -
+ - You should have received a copy of the GNU Affero General Public License
+ - along with this program. If not, see <http://www.gnu.org/licenses/>.
+ -
+ -->
+<docs>
+Just nest the `AppSettingSections` component into `AppSettingsDialog`,
+providing the section's title prop. You can put your settings within each
+`AppSettingsSection` component.
+
+```vue
+<template>
+	<div>
+		<button @click="settingsOpen = true">Show Settings</button>
+		<AppSettingsDialog :open.sync="settingsOpen" >
+			<AppSettingsSection title="Example title 1">
+				Some example content
+			</AppSettingsSection>
+			<AppSettingsSection title="Example title 2">
+				Some more content
+			</AppSettingsSection>
+		</AppSettingsDialog>
+	</div>
+</template>
+
+<script>
+export default {
+	data() {
+		return {
+			settingsOpen: false,
+		}
+	},
+}
+</script>
+```
+</docs>
+
+<script>
+import Modal from '../Modal'
+import isMobile from '../../mixins/isMobile'
+import debounce from 'debounce'
+
+export default {
+
+	name: 'AppSettingsDialog',
+
+	components: {
+		Modal,
+	},
+
+	mixins: [isMobile],
+
+	props: {
+		/**
+		 * Determines the open / closed state of the modal
+		 */
+		open: {
+			type: Boolean,
+			required: true,
+		},
+		/**
+		 * Hides the navigation on desktop if true
+		 */
+		hideNavigation: {
+			type: Boolean,
+			default: false,
+		},
+	},
+
+	data() {
+		return {
+			selectedSection: '',
+			linkClicked: false,
+			addedScrollListener: false,
+			scroller: null,
+		}
+	},
+
+	computed: {
+
+		hasNavigation() {
+			if (this.isMobile || this.hideNavigation) {
+				return false
+			} else {
+				return true
+			}
+		},
+
+		settingsNavigationAriaLabel() {
+			return t('Settings navigation')
+		},
+	},
+
+	mounted() {
+		// Select first settings section
+		this.selectedSection = this.$slots.default[0].componentOptions.propsData.title
+	},
+
+	updated() {
+		if (!this.scroller) {
+			// Get the scroller element
+			this.scroller = document.querySelector('.app-settings__content')
+		}
+		if (this.scroller && !this.addedScrollListener) {
+			this.scroller.addEventListener('scroll', debounce(() => {
+				this.handleScroll()
+			}), 300)
+			this.addedScrollListener = true
+		}
+	},
+
+	beforeDestroy() {
+		document.removeEventListener('scroll', this.handleScroll)
+	},
+	methods: {
+
+		/**
+		 * Builds the settings navigation menu
+		 * @param {object} slots The default slots object passed from the render function.
+		 * @returns {array} the navigation items
+		 */
+		getSettingsNavigation(slots) {
+			// Array of navigationitems strings
+			const navigationItems = slots.filter(vNode => vNode.componentOptions).map(vNode => vNode.componentOptions.propsData.title)
```

Comment:
Please add optional chaining to be safe

Reply:
what do you mean here @skjnldsv ??

Reply:
https://duckduckgo.com/?q=optional+chaining&t=brave&ia=web

Reply:
Can you show me exactly where you'd put it? I'm afraid I can't see the problem you're trying to solve

Reply:
Well you're filtering by chaining multiple objects properties.
Make sure everything is defined